### PR TITLE
feat: update spin extension to v0.14.1

### PR DESCRIPTION
**Pull request by rgl**

### CODE REVIEW

File: container-runtime/vars.yaml [hunk 1]
```diff
@@ -7,6 +7,6 @@ CLOUD_PROVIDER_AWS_VERSION: v1.30.0
 # renovate: datasource=github-tags depName=containerd/runwasi
 WASMEDGE_VERSION: v0.3.0
 # renovate: datasource=github-tags depName=spinkube/containerd-shim-spin
-SPIN_VERSION: v0.13.1
+SPIN_VERSION: v0.14.1
```
Comment (frezbo) [inline container-runtime/vars.yaml:10]: this needs updating checksums too https://github.com/siderolabs/extensions/blob/main/container-runtime/spin/pkg.yaml#L9

Comment (rgl) [reply container-runtime/vars.yaml:10]: oh, indeed. check it now please.

**Review by frezbo — APPROVED** [review]

**Pull request merged**